storage/series/series.go

```diff
@@ -706,3 +735,18 @@ func (s *dbSeries) Reset(
 	s.onRetrieveBlock = onRetrieveBlock
 	s.blockOnEvictedFromWiredList = onEvictedFromWiredList
 }
+
+// FlushOutcome is an enum that provides more context about the outcome
+// of series.Flush() to the caller.
+type FlushOutcome int
+
+const (
+	// FlushErr is just a default value that can be returned when we're also returning an error
+	FlushErr FlushOutcome = iota
+	// BlockDoesNotExist indicates that the series did not have a block for the specified flush blockStart.
+	BlockDoesNotExist
+	// StreamDoesNotExist indicates that the block did not have an underlying stream.
+	StreamDoesNotExist
```

Comment:
This is probably an error btw, maybe we should just treat it as such.

Reply:
done